pyccel/parser/semantic.py

```diff
@@ -964,6 +964,21 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, new_expressions, is_augassign, *
                     if 'allow_negative_index' in decorators:
                         if lhs in decorators['allow_negative_index']:
                             d_lhs.update(allows_negative_indexes=True)
+                
+                # We cannot allow the definition of a stack array from a shape with non-literal integers
```

Comment:
This comment line is confusing, see previous comment

Reply:
Fixed [98096de](https://github.com/pyccel/pyccel/pull/993/commits/98096ded80323407e4a8468c0e0d69747973bc46)